### PR TITLE
Add ZK-3722 test case

### PR DESCRIPTION
**Pull request by klyvechen**

### CODE REVIEW

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+				chechPos(num, offleft, offTop)
+				if (num == 4 || num == 5) {
+					checkWidth(i.toString)
+				}
+			}
+			verifyTrue(getEval("jq('.z-column:contains(\"Apr\"):eq(0)').get(0).style.visibility").equals("hidden"))
+			verifyTrue(getEval("jq('.z-column:contains(\"May\"):eq(0)').get(0).style.visibility").equals("hidden"))
+			verifyTrue(getEval("jq('.z-column:contains(\"Apr\"):eq(1)').get(0).style.visibility").equals("hidden"))
+			verifyTrue(getEval("jq('.z-column:contains(\"May\"):eq(1)').get(0).style.visibility").equals("hidden"))
+		})
```
Comment (DevChu) [inline zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: why getEval

Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+			MayPos1 = getEval("jq('.z-column:contains(\"May\"):eq(1)').offset().left").toDouble
+			JunPos1 = getEval("jq('.z-column:contains(\"Jun\"):eq(1)').offset().left").toDouble
+			JulPos1 = getEval("jq('.z-column:contains(\"Jul\"):eq(1)').offset().left").toDouble
+			AugPos1 = getEval("jq('.z-column:contains(\"Aug\"):eq(1)').offset().left").toDouble
+			SepPos1 = getEval("jq('.z-column:contains(\"Sep\"):eq(1)').offset().left").toDouble
+			OctPos1 = getEval("jq('.z-column:contains(\"Oct\"):eq(1)').offset().left").toDouble
+			NovPos1 = getEval("jq('.z-column:contains(\"Nov\"):eq(1)').offset().left").toDouble
+			DecPos1 = getEval("jq('.z-column:contains(\"Dec\"):eq(1)').offset().left").toDouble
+
+//			JanPos1 = jq(".z-column:contains('Jan'):eq(1)").offsetLeft()
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+package org.zkoss.zktest.test2.B85
+
+import org.junit.Test
+import org.zkoss.zstl.ZTL4ScalaTestCase
+import org.zkoss.ztl.Tags
+
+@Tags(tags = "")
+class B85_ZK_3722Test extends ZTL4ScalaTestCase {
+	var cellSize = 0
+	var rowinnerSize = 0
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+				waitResponse()
+				chechPos(num, offleft, offTop)
+				if (num == 4 || num == 5) {
+					checkWidth(i.toString)
+				}
+			}
+			verifyTrue(jq(".z-column:contains(\"Apr\"):eq(0)").get(0).eval("style.visibility").equals("hidden"))
+			verifyTrue(jq(".z-column:contains(\"May\"):eq(0)").get(0).eval("style.visibility").equals("hidden"))
+			verifyTrue(jq(".z-column:contains(\"Apr\"):eq(1)").get(0).eval("style.visibility").equals("hidden"))
+			verifyTrue(jq(".z-column:contains(\"May\"):eq(1)").get(0).eval("style.visibility").equals("hidden"))
```
Comment (scribetw) [inline zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]:
```
verifyEquals("hidden", jq(".z-column:contains(\"May\"):eq(1)").css("visibility"))
```

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		})
+	}
+
+	def chechPos(num : Int, offleft : Double, offTop: Double)  = {
```
Comment (scribetw) [inline zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: check

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		System.out.println(num, monList.apply(num - 1))
+		if (offTop < grid2Top) {
+			targetPos = pos0Map.apply(monList.apply(num - 1))
+		} else {
+			targetPos = pos1Map.apply(monList.apply(num - 1))
+		}
+		verifyTrue((offleft- targetPos) < 1.0 | (offleft- targetPos) > -1.0)
+	}
+	def checkWidth(i : String) = {
+		verifyTrue(getEval("jq(jq('.z-row-inner').get(" + i + ")).width()").equals("0"))
```
Comment (scribetw) [inline zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]:
```
verifyEquals(0, jq(".z-row-inner").eq(i).width())
```

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+			rowinnerSize = jq(".z-row-inner").length()
+			grid2Top = jq(".z-grid:eq(1)").offsetTop()
+			for (i <- 0 to 11) {
+				var mon = monList.apply(i)
+				pos0Map += (mon -> jq(".z-column:contains(\""+ mon +"\"):eq(0)").offsetLeft())
+				pos1Map += (mon -> jq(".z-column:contains(\""+ mon +"\"):eq(1)").offsetLeft())
+			}
+			waitResponse()
+			for (i <- 0 to cellSize - 1) {
+				var num = jq(jq(".z-cell").get(i)).find("span").text().toInt
```
Comment (scribetw) [inline zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: `jq(".z-cell").eq(i).find(...)....`

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+	def chechPos(num : Int, offleft : Double, offTop: Double)  = {
+		var targetPos = 0.0
+		targetPos = pos0Map.apply(monList.apply(num - 1))
+		System.out.println(num, monList.apply(num - 1))
+		if (offTop < grid2Top) {
+			targetPos = pos0Map.apply(monList.apply(num - 1))
+		} else {
+			targetPos = pos1Map.apply(monList.apply(num - 1))
+		}
+		verifyTrue((offleft- targetPos) < 1.0 | (offleft- targetPos) > -1.0)
```
Comment (scribetw) [inline zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: Use `verifyTolerant(a, b, delta)`, set delta = 1 or 2

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala (hunk 7 above)
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala (hunk 6 above)
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala (hunk 5 above)
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala (hunk 4 above)
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

File: zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala (hunk 8 above)
Comment (klyvechen) [reply zstl/test/org/zkoss/zktest/test2/B85/B85_ZK_3722Test.scala]: ok

**Pull request merged**